service/service.go

```diff
@@ -2,162 +2,203 @@ package service
+	}
+
+	svc.cfg = cfg
+
+	// Get Kafka consumer
+	if svc.consumer, err = GetKafkaConsumer(ctx, cfg); err != nil {
+		return fmt.Errorf("failed to initialise kafka consumer: %w", err)
+	}
 
 	// Get HealthCheck
-	hc, err := serviceList.GetHealthCheck(cfg, buildTime, gitCommit, version)
-	if err != nil {
-		log.Event(ctx, "could not instantiate healthcheck", log.FATAL, log.Error(err))
-		return nil, err
+	if svc.healthCheck, err = GetHealthCheck(cfg, buildTime, gitCommit, version); err != nil {
+		return fmt.Errorf("could not instantiate healthcheck: %w", err)
 	}
 
-	if err := registerCheckers(ctx, hc, consumer); err != nil {
-		return nil, errors.Wrap(err, "unable to register checkers")
+	if err := svc.registerCheckers(); err != nil {
+		return fmt.Errorf("unable to register checkers: %w", err)
 	}
 
-	r.StrictSlash(true).Path("/health").HandlerFunc(hc.Handler)
-	hc.Start(ctx)
+	// Get HTTP Server with collectionID checkHeader
+	r := mux.NewRouter()
+	r.StrictSlash(true).Path("/health").HandlerFunc(svc.healthCheck.Handler)
+	svc.server = GetHTTPServer(cfg.BindAddr, r)
+
+	return nil
+}
+
+// Start starts an initialised service
+func (svc *Service) Start(ctx context.Context, svcErrors chan error) {
+	log.Info(ctx, "starting service...")
+
+	// Start kafka error logging
+	svc.consumer.Channels().LogErrors(ctx, "error received from kafka consumer, topic: "+svc.cfg.HelloCalledTopic)
+
+	// Start consuming Kafka messages with the Event Handler
+	event.Consume(
+		ctx,
+		svc.consumer,
+		event.NewHelloCalledHandler(
+			*svc.cfg,
+		),
+		svc.cfg.KafkaNumWorkers,
+	)
+
+	// Start health checker
+	svc.healthCheck.Start(ctx)
 
 	// Run the http server in a new go-routine
 	go func() {
-		if err := s.ListenAndServe(); err != nil {
+		if err := svc.server.ListenAndServe(); err != nil {
 			svcErrors <- errors.Wrap(err, "failure in http listen and serve")
 		}
 	}()
-
-	return &Service{
-		server:          s,
-		router:          r,
-		serviceList:     serviceList,
-		healthCheck:     hc,
-		consumer:        consumer,
-		shutdownTimeout: cfg.GracefulShutdownTimeout,
-	}, nil
 }
 
 // Close gracefully shuts the service down in the required order, with timeout
 func (svc *Service) Close(ctx context.Context) error {
-	timeout := svc.shutdownTimeout
-	log.Event(ctx, "commencing graceful shutdown", log.Data{"graceful_shutdown_timeout": timeout}, log.INFO)
+	timeout := svc.cfg.GracefulShutdownTimeout
+	log.Info(ctx, "commencing graceful shutdown", log.Data{"graceful_shutdown_timeout": timeout})
 	ctx, cancel := context.WithTimeout(ctx, timeout)
-
-	// track shutdown gracefully closes up
-	var gracefulShutdown bool
+	hasShutdownError := false
 
 	go func() {
 		defer cancel()
-		var hasShutdownError bool
 
 		// stop healthcheck, as it depends on everything else
-		if svc.serviceList.HealthCheck {
+		if svc.healthCheck != nil {
 			svc.healthCheck.Stop()
+			log.Event(ctx, "stopped health checker", log.INFO)
```

Comment:
Opportunity to use log.Info rather than log.Event

Reply:
thanks! I thought I had replaced all of them